gluecodium/src/test/resources/smoke/attributes/output/android/com/example/smoke/AttributesClass.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *
+ */
+package com.example.smoke;
+import android.support.annotation.NonNull;
+import com.example.NativeBase;
+@OnClass
+public final class AttributesClass extends NativeBase {
+    @OnConstInClass
+    public static final boolean PI = false;
+    /**
+     * For internal use only.
+     * @exclude
+     */
+    protected AttributesClass(final long nativeHandle, final Object dummy) {
+        super(nativeHandle, new Disposer() {
+            @Override
+            public void disposeNative(long handle) {
+                disposeNativeHandle(handle);
+            }
+        });
+    }
+    private static native void disposeNativeHandle(long nativeHandle);
+    @OnFunctionInClass
+    public native void veryFun(@OnParameterInClass @NonNull final String param);
+    @OnPropertyInClass
+    @NonNull
```

Comment:
I guess `@NonNull`, `@Deprecated`, etc can be switched to the same mechanism, it may simplify mustache files

Reply:
No. Nullability in it's current form is up to standard of the modern languages (e.g. Kotlin, Swift, Dart). Shifting away from `?` syntax to ultra-verbose attribute syntax sounds bad on its own. "Simplifying templates" at the cost of forcing the user to use ridiculous amount of boilerplate is just a crime.

Similarly, current `@Deprecated` applies to all platforms automatically. Moving away from it multiplies boilerplate x4. So, no. On the bright side, the custom attributes allow for per-platform deprecation... but I need to make sure escaped quotes `\"` are processed correctly.

Reply:
I don't mean to change anything in lime syntax, I'm talking only about internal implementation in Gluecodium. Gluecodium may put `@Deprecated` and `@NonNull` to internal collection of attributes and mustache will have only one placeholder `{{>java/JavaAttributes}}`

Reply:
Sounds doable then. But I'm still not sure it's worth it. For example, `@NonNull` is Java-specific, so it would require some additional post-processing of LIME model specifically for Java generator.